--- a/obj/auditor.py	2014-04-18 17:13:45.000000000 +0800
+++ b/obj/auditor.py	2014-08-25 20:53:37.756333411 +0800
@@ -22,6 +22,7 @@
 from eventlet import Timeout
 
 from swift.obj import diskfile
+from vs_auditor import vs_diskfile
 from swift.common.utils import get_logger, ratelimit_sleep, dump_recon_cache, \
     list_from_csv, json, listdir
 from swift.common.exceptions import DiskFileQuarantined, DiskFileNotExist
@@ -41,6 +42,11 @@
         self.max_files_per_second = float(conf.get('files_per_second', 20))
         self.max_bytes_per_second = float(conf.get('bytes_per_second',
                                                    10000000))
+        
+        # the following lines are added to support ValueStore specific
+        # VSDiskFileManager, nothing else is changed in the auditor
+        self.diskfile_mgr = vs_diskfile.VSDiskFileManager(conf, self.logger)
+        
         self.auditor_type = 'ALL'
         self.zero_byte_only_at_fps = zero_byte_only_at_fps
         if self.zero_byte_only_at_fps:

